fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/ClientRepository.java

```diff
@@ -29,4 +29,35 @@ interface ClientRepository extends JpaRepository<Client, Long>, JpaSpecification
 
     @Query(FIND_CLIENT_BY_ACCOUNT_NUMBER)
     Client getClientByAccountNumber(@Param("accountNumber") String accountNumber);
+
+    @Query("""
+            SELECT client.id
+            FROM Client client
+            WHERE client.externalId = :externalId
+                """)
+    Long getClientIdByExternalId(String externalId);
+
+    @Query("""
+            SELECT client
+            FROM Client client
+            JOIN client.office office
+            LEFT JOIN client.transferToOffice transferToOffice
+            WHERE client.id = :clientId
+            AND (office.hierarchy LIKE :officeHierarchy OR transferToOffice.hierarchy LIKE :transferToOfficeHierarchy)
+                """)
+    Client fetchByClientIdAndHierarchy(Long clientId, String officeHierarchy, String transferToOfficeHierarchy);
+
+    @Query("""
+            SELECT client
+            FROM Client client
+            JOIN client.office office
+            LEFT JOIN client.transferToOffice transferToOffice
+            WHERE client.externalId = :externalId
+            AND (office.hierarchy LIKE :officeHierarchy OR transferToOffice.hierarchy LIKE :transferToOfficeHierarchy)
+                """)
+    Client fetchByExternalIdAndHierarchy(String externalId, String officeHierarchy, String transferToOfficeHierarchy);
+
+    @Query("SELECT c.id FROM Client c WHERE c.externalId = :externalId")
+    Long findIdByExternalId(@Param("externalId") String externalId);
```

Comment:
What is the difference between this method and "getClientIdByExternalId" ?

Reply:
`getClientIdByExternalId` was removed